lib/src/components/comms_button/comms_button.dart

```diff
@@ -0,0 +1,431 @@
+    super.key,
+    this.label,
+    this.size = ZetaWidgetSize.medium,
+    this.onPressed,
+    this.focusNode,
+    this.semanticLabel,
+  })  : type = ZetaCommsButtonType.warning,
+        icon = ZetaIcons.alert_active,
+        onToggle = null,
+        toggledIcon = null,
+        toggledLabel = null,
+        toggledType = null;
+
+  /// Comms button label
+  final String? label;
+
+  /// Called when the comms button is toggled.
+  /// If null, the comms button will not be toggleable.
+  final ValueChanged<bool>? onToggle;
+
+  /// Icon to display when the comms button is toggled.
+  final IconData? toggledIcon;
+
+  /// Label to display when the comms button is toggled.
+  /// If null, the [label] will be used instead.
+  /// If both [label] and [toggledLabel] are null, the comms button will not display a label.
+  final String? toggledLabel;
+
+  /// The coloring type of the comms button when toggled.
+  /// Defaults to [ZetaCommsButtonType.on].
+  final ZetaCommsButtonType? toggledType;
+
+  /// Called when the comms button is tapped or otherwise activated.
+  ///
+  /// {@macro zeta-widget-change-disable}
+  final VoidCallback? onPressed;
+
+  /// The coloring type of the comms button
+  final ZetaCommsButtonType type;
+
+  /// Size of the comms button. Defaults to [ZetaWidgetSize.medium].
+  final ZetaWidgetSize size;
+
+  /// Icon of comms button. Goes in centre of button.
+  final IconData? icon;
+
+  /// {@macro flutter.widgets.Focus.focusNode}
+  final FocusNode? focusNode;
+
+  /// The semantic label of the comms button.
+  ///
+  /// {@macro zeta-widget-semantic-label}
+  ///
+  /// If this property is null, [label] or [toggledLabel] will be used instead.
+  final String? semanticLabel;
+
+  @override
+  State<ZetaCommsButton> createState() => _ZetaCommsButtonState();
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties
+      ..add(StringProperty('label', label))
+      ..add(ObjectFlagProperty<VoidCallback?>.has('onPressed', onPressed))
+      ..add(ObjectFlagProperty<ValueChanged<bool>?>.has('onToggle', onToggle))
+      ..add(ObjectFlagProperty<IconData?>.has('toggledIcon', toggledIcon))
+      ..add(StringProperty('toggledLabel', toggledLabel))
+      ..add(EnumProperty<ZetaCommsButtonType>('type', type))
+      ..add(EnumProperty<ZetaCommsButtonType>('toggledType', toggledType))
+      ..add(EnumProperty<ZetaWidgetSize>('size', size))
+      ..add(DiagnosticsProperty<IconData?>('icon', icon))
+      ..add(DiagnosticsProperty<FocusNode?>('focusNode', focusNode))
+      ..add(StringProperty('semanticLabel', semanticLabel));
+  }
+}
+
+class _ZetaCommsButtonState extends State<ZetaCommsButton> {
+  bool isToggled = false;
+  late ZetaCommsButtonType type;
+  late Color iconColor = _iconColor(context, type);
+  late Color backgroundColor = _backgroundColor(context, type);
+  late Color borderColor = _borderColor(context, type);
```

Comment:
are these variables up here being used as state values? I think they can just be moved into the build method?